app/vmui/packages/vmui/src/hooks/useFetchQueryOptions.tsx

```diff
@@ -5,140 +5,168 @@ import { AutocompleteOptions } from "../components/Main/Autocomplete/Autocomplet
+import { AUTOCOMPLETE_LIMITS } from "../components/Configurators/QueryEditor/QueryAutocompleteCache";
 
 enum TypeData {
-  metric,
-  label,
-  value
+  metric = "metric",
+  label = "label",
+  value = "value"
 }
 
 type FetchDataArgs = {
+  value: string;
   urlSuffix: string;
   setter: StateUpdater<AutocompleteOptions[]>;
   type: TypeData;
   params?: URLSearchParams;
 }
 
+type FetchQueryArguments = {
+  valueByContext: string;
+  metric: string;
+  label: string;
+  context: QueryContextType
+}
+
 const icons = {
-  [TypeData.metric]: <MetricIcon />,
-  [TypeData.label]: <LabelIcon />,
-  [TypeData.value]: <ValueIcon />,
+  [TypeData.metric]: <MetricIcon/>,
+  [TypeData.label]: <LabelIcon/>,
+  [TypeData.value]: <ValueIcon/>,
 };
 
-const QUERY_LIMIT = 1000;
-
-export const useFetchQueryOptions = ({ metric, label }: { metric: string; label: string }) => {
+export const useFetchQueryOptions = ({ valueByContext, metric, label, context }: FetchQueryArguments) => {
   const { serverUrl } = useAppState();
   const { period: { start, end } } = useTimeState();
+  const { autocompleteCache } = useQueryState();
+  const queryDispatch = useQueryDispatch();
+
+  const [loading, setLoading] = useState(false);
+  const [value, setValue] = useState(valueByContext);
+  const debouncedSetValue = debounce(setValue, 800);
+  useEffect(() => {
+    debouncedSetValue(valueByContext);
+    return debouncedSetValue.cancel;
+  }, [valueByContext, debouncedSetValue]);
 
   const [metrics, setMetrics] = useState<AutocompleteOptions[]>([]);
   const [labels, setLabels] = useState<AutocompleteOptions[]>([]);
   const [values, setValues] = useState<AutocompleteOptions[]>([]);
 
-  const prevParams = useRef<Record<string, URLSearchParams>>({});
+  const abortControllerRef = useRef(new AbortController());
 
   const getQueryParams = useCallback((params?: Record<string, string>) => {
-    const roundedStart = dayjs(start).startOf("day").valueOf();
-    const roundedEnd = dayjs(end).endOf("day").valueOf();
-
     return new URLSearchParams({
       ...(params || {}),
-      limit: `${QUERY_LIMIT}`,
-      start: `${roundedStart}`,
-      end: `${roundedEnd}`
+      limit: `${AUTOCOMPLETE_LIMITS.queryLimit}`,
+      start: `${start}`,
+      end: `${end}`
     });
   }, [start, end]);
 
-  const isParamsEqual = (prev: URLSearchParams, next: URLSearchParams) => {
-    const queryNext = qs.parse(next.toString());
-    const queryPrev = qs.parse(prev.toString());
-    return JSON.stringify(queryPrev) === JSON.stringify(queryNext);
+  const processData = (data: string[], type: TypeData) => {
+    return data.map(l => ({
+      value: l,
+      type: `${type}`,
+      icon: icons[type]
+    }));
   };
 
-  const fetchData = async ({ urlSuffix, setter, type, params }: FetchDataArgs) => {
+  const fetchData = async ({ value, urlSuffix, setter, type, params }: FetchDataArgs) => {
+    abortControllerRef.current.abort();
+    abortControllerRef.current = new AbortController();
```

Comment:
After calling `AbortController.signal.abort()`, the `AbortController` cannot be reused, so a new one must be created. 
In this case, the `AbortController` is needed to aborted a request that is no longer relevant to the user because they now require different values for suggestions.